factcast-factus/src/test/java/org/factcast/factus/applier/ComplexAggregate.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright © 2017-2020 factcast.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.factcast.factus.applier;
+
+import java.util.UUID;
+
+import org.factcast.factus.Handler;
+import org.factcast.factus.projection.Aggregate;
+import org.factcast.factus.projection.AggregateUtil;
+
+import lombok.Getter;
+
+class ComplexAggregate extends Aggregate {
+
+    ComplexAggregate(UUID aggregateId) {
+        AggregateUtil.aggregateId(this, aggregateId);
```

Comment:
changed the setter to protected & added an AllArgsConstr., so that we can call it here. going over Util does not feel right.